### PR TITLE
EVL-217: list access_control groups in GET /users/groups

### PR DESCRIPTION
**Pull request by p3rotto**
## 📝 Description

`GET /users/groups` read only `private.user_groups`. That table is a membership
table, not a group catalog: a row exists only while a user, service account, API
key or AI agent is bound to the group.

IdP sync (OIDC and SAML) deletes and reinserts a user's rows on every login. When
the last Hoop member of a group stopped carrying it, every row disappeared, so
the group vanished from the endpoint. Its `access_control` association survived
and kept applying at connection time, so the permission stayed live but became
unreachable in the UI.

The endpoint now unions the identity side with the group names referenced by the
`access_control` plugin config, and sorts the result. The MCP `usergroups_list`
tool gets the same union so the two surfaces cannot disagree.

## 🎯 User-facing impact

Admins can again assign a group to an API key when that group only exists as an
`access_control` permission. Same for AI Agent identities, Roles → Configure,
Access Request, and agents using the MCP tool.

Two smaller fixes ride along: the list was in random order (Go map iteration), and
an org with no groups serialized as `null` instead of `[]`.

## 🔗 Related Issue

EVL-217

## 🔄 Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🎨 Style/UI update
- [ ] ♻️ Refactor (no functional changes)
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test update
- [ ] 🔧 Configuration change

## 📋 Changes Made

- `gateway/models/plugin_connections.go`: new `ListAccessControlGroupNames`,
  reading the distinct group names from `plugin_connections.config` for the
  `access_control` plugin. The predicate deliberately mirrors the enforcement
  queries in `GetConnectionByNameOrID` and `GetResourceByName`: join on
  `plugin_id`, scope the org through `private.plugins`. It does **not** filter
  `pc.enabled`, because enforcement does not either, and it scopes by `p.org_id`
  because `pc.org_id` is nullable. Narrowing it would hide a group whose
  permissions still apply, which is the bug being fixed.
- `gateway/api/user/user.go`: `ListAllGroups` unions both sources, sorts with
  `slices.Sort`, and returns an empty slice instead of `nil`.
- `gateway/api/mcpserver/tools_usergroups.go`: `usergroups_list` gets the same
  union. Without it an agent cannot see a group that exists only as a permission,
  and may call `usergroups_create` on a name that already grants access.
- `webapp_v2/src/pages/Roles/Configure/components/ReviewSection.jsx`: the approver
  seed was `userGroupsList[0]`, which was `admin` only because the list happened to
  be unsorted. Sorting plus the union made position 0 a group that may have no
  members. The seed now comes from the current user's own groups intersected with
  the available list, preferring `admin`. The route is `adminOnly`, so that
  intersection is never empty.
- `gateway/models/plugin_connections_test.go`: pglite tests covering dedup, a
  disabled association (listed, with the reason), another plugin's config, a null
  config, and an org with no `access_control` plugin.
- Comment and doc corrections for the `null` → `[]` change:
  `webapp_v2/COMPONENTS.md`, `webapp_v2/src/services/userGroups.js`,
  `webapp_v2/src/pages/Features/AccessRequest/store.js`,
  `webapp_v2/src/pages/Features/AccessControl/helpers.js`.
- OpenAPI regenerated for the new endpoint description.

### Test Configuration

- Go 1.26.5, `make test-oss` (exit 0, zero failures)
- Model tests run against embedded PGlite with the full migration set

### Tests performed

- [x] Unit tests pass locally
- [x] Integration tests pass locally
- [ ] Manual testing completed
- [x] Tested with different user roles
- [x] Tested edge cases

`TestListAccessControlGroupNames` covers two connections sharing a group (deduped),
a **disabled** association (must be listed: enforcement ignores the column), an
`audit` plugin config (excluded), and a null config (no crash).
`TestListAccessControlGroupNamesWithoutPlugin` seeds an `audit` row so the
plugin-name filter is load-bearing rather than passing vacuously.

Role coverage is by route: `GET /users/groups` is `ReadOnlyAccessRole`, unchanged.

**How to test manually**

1. Start the gateway with a fresh DB, log in as admin.
2. Create a connection and go to Access Control → New group. Name it `test-orphan`
   and attach it to that connection. Save.
3. Confirm `test-orphan` appears at Settings → API Keys → Create → Groups.
4. Reproduce the customer state: delete the identity-side row only, leaving the
   plugin association intact.
   `DELETE FROM private.user_groups WHERE org_id = '<org>' AND name = 'test-orphan';`
5. Reload Settings → API Keys → Create. Before this PR the group is gone from the
   picker; with this PR it is still listed, and selecting it saves normally.
6. Check `GET /api/users/groups` returns the list sorted, and `[]` (not `null`) on
   an org with no groups.

## 📸 Screenshots

Not applicable. No UI change; the pickers are fed by the endpoint.

## ✅ Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

## 📌 Additional Notes

**Customer workaround while this ships.** `POST /api/users/groups {"name":"<group>"}`
writes a standalone row with a null `user_id`, so the group reappears in every
picker without touching permissions. Sent to the customer on EVL-217.

**Follow-ups found during the investigation, not fixed here:**

1. EVL-221 - the access_control write path never creates the standalone
   `user_groups` row, which is the invariant this PR works around on the read
   side. Also closes an asymmetry: MCP `usergroups_delete` hard-errors on a
   plugin-only group while the webapp swallows the same 404.
2. EVL-223 - the reported screenshot shows two pills both reading
   `CG-Hoop-FINANCIAL-SERVICES-PF` on one API key. The partial unique index
   `idx_user_groups_api_key_name` on `(api_key_id, name)` makes an exact duplicate
   impossible, so the two names differ invisibly, most likely by trailing
   whitespace. Nothing in the sync path trims group names, and a stray space
   silently breaks the `config && user_groups` overlap test. Asked the customer to
   confirm.
3. EVL-222 - no handling for Entra ID group-claim overage (`_claim_names` /
   `_claim_sources`). A user in more than 150 SAML or 200 JWT groups receives no
   groups at all. Same symptom as this bug from the user's side.
4. EVL-224 - `gateway/idp/saml/saml.go:90-92` computes the `groups_claim` default
   and uses it only in the log line at 126, so the provider keeps the raw value.
   Latent, not live: `gateway/api/serverconfig/auth.go:341-342` already defaults
   the field before the row is written, and `newSamlProvider` reads only from that
   row. The log line still reports a default that is not applied.
5. EVL-225 - `admin_role_name` is absent from `/serverinfo`, so `webapp_v2`
   hardcodes `'admin'` in three places.
6. `private.user_groups` has no index on `org_id`, so `GetUserGroupsByOrgID` is a
   sequential scan returning one row per binding. Not a regression, and the query
   added here is smaller, but it is the dominant cost of this endpoint. No ticket.

### CODE REVIEW

**Comment by sandromello** [comment]
✅ Build Completed with Success, Version=1742.0.0-g5d3bd97

**Comment by sandromello** [comment]
✅ Build Completed with Success, Version=1742.0.0-g6fa9be9

**Comment by sandromello** [comment]
✅ Build Completed with Success, Version=1742.0.0-g2977ee5

**Comment by sandromello** [comment]
✅ Build Completed with Success, Version=1742.0.0-ge50a6cc

**Comment by sandromello** [comment]
✅ Build Completed with Success, Version=1742.0.0-g82bc58c

File: webapp_v2/src/pages/Roles/Configure/components/ReviewSection.jsx [hunk 1]
```diff
@@ -37,10 +37,13 @@ export default function ReviewSection({ kind = 'command' }) {
...
     if (enabled) {
-      // Surfacing the toggle without any approval groups would be invalid
-      // — seed with the first available group so the section is usable.
+      // Surfacing the toggle without any approval groups would be invalid, so
+      // seed one. Prefer admin: /users/groups is sorted since EVL-217, so
+      // position 0 is now whichever group sorts first and would hand approval
+      // to an arbitrary customer group if the user saves without editing.
       if (drafts.reviewers.length === 0 && userGroupsList.length > 0) {
-        setDraft({ reviewers: [userGroupsList[0]] })
+        const seed = userGroupsList.includes('admin') ? 'admin' : userGroupsList[0]
```
[the remark this answers is not in the document]
Comment (p3rotto) [reply webapp_v2/src/pages/Roles/Configure/components/ReviewSection.jsx]: ✅ **Fixed** — commit `12be4eaa`. Seeds the reviewer from the current admin's own groups (this route is `adminOnly`), so the memberless access_control-only group this PR could surface is never selected. `admin` stays a preference, not an assumption. Not adopting fix PR #1746: seeding `null` leaves the toggle dead, since `reviewEnabled` derives from `reviewers.length > 0`.

File: gateway/api/mcpserver/tools_usergroups.go [hunk 2]
```diff
@@ -61,6 +62,13 @@ func usergroupsListHandler(ctx context.Context, _ *mcp.CallToolRequest, _ usergr
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed listing user groups: %w", err)
 	}
+	// Same union as GET /users/groups, or an agent cannot see a group that only
+	// exists as an access_control permission and calls usergroups_create on a
+	// name that already grants access (EVL-217).
```
[the remark this answers is not in the document]
Comment (p3rotto) [reply gateway/api/mcpserver/tools_usergroups.go:67]: ⏭️ **Deferred** — the repo convention is the opposite of rule 2362449: 66 Go comments reference a ticket parenthetically mid-sentence (`DEP-48`, `ENG-396`, `DEP-16`) against 3 that prefix it. `CLAUDE.md` requires ID-not-URL only, which these comments already satisfy. Updating the rule to match the codebase instead of rewriting the 7 comments.

**Comment by p3rotto** [comment]
## Qodo Fix Summary — Round 1

### ☑ Fixed (1)

**Configured admin seed ignored** — `webapp_v2/src/pages/Roles/Configure/components/ReviewSection.jsx:45` (commit `12be4eaa`)

The seed now comes from the current user's own groups, filtered against the available list, preferring `admin`. This route is `adminOnly` (`Router.jsx:131`), so those groups always include the administrator group under whatever name it carries. Fixes the real risk this PR introduced: the union added access-control-only groups, so `userGroupsList[0]` could be a group with no members that nobody can approve with.

Not adopting auto-fix PR #1746. Seeding `null` when `admin` is absent leaves the toggle dead with no feedback, because `reviewEnabled` is derived from `reviewers.length > 0`. It also keeps the `admin` literal and adds an unused `useEffect, useState` import.

Reading the configured name properly needs `admin_role_name` in `ServerInfo` — it exists only in the admin-only `/serverconfig/auth` response today. That is a gateway change outside EVL-217; tracked as a follow-up.

### ⏭ Deferred (1)

**EVL-217 comments lack prefix** — 7 sites, rule 2362449

The rule contradicts this repository. 66 Go comments reference a ticket parenthetically mid-sentence (`DEP-48`, `ENG-396`, `DEP-16`); 3 prefix it. `CLAUDE.md` requires ID-not-URL only, which every comment in this PR satisfies. Rule 2362449 is being updated to match the codebase.

---

_Generated by Qodo PR Resolver skill_

**Review by matheusfrancisco — APPROVED** [review]

**Comment by sandromello** [comment]
✅ Build Completed with Success, Version=1742.0.0-g12be4ea
